crates/wasi/tests/all/sync.rs

```diff
@@ -294,6 +294,14 @@ fn preview2_stream_pollable_traps() {
     )
 }
 #[test_log::test]
+fn preview2_pollable_correct() {
+    run(PREVIEW2_POLLABLE_CORRECT_COMPONENT, false).unwrap()
+}
+#[test_log::test]
+fn preview2_pollable_traps() {
+    run(PREVIEW2_POLLABLE_TRAPS_COMPONENT, false).unwrap_err();
```

Comment:
There's quite a lot of errors which go through the `Err` here so I'm a bit wary of using `unwrap_err` to assert that a particular trap happened, could this instead a particular property about the error, such as its message?

Reply:
Certainly